### PR TITLE
Added support for multiple oracle pools

### PR DESCRIPTION
**Pull request by cemremengu**
This PR adds support for multiple pool definitions.

*Changes*
- Options now support a `name` property for pool labeling (note that this is not `poolAlias`). Labeled clients can be accessed via `oracle.<dbname>.getConnection() `
- `oracledb` instance is exposed as `oracle.db` so that you can access constants and other options such as constants like `oracle.db.OBJECT`
- (*breaking*) `getConnection` is preserved but the pool instance is now accessed via `oracle.pool`

```js
fastify.register(require("fastify-oracle"), {
pool: {
  user: "oracle",
  password: "oracle",
  connectString: "oracle.local:1521/mydb"
},
name: 'mydb'
})

async function (request, reply) {
 // Or this.oracle.mydb.getConnection()
  const conn = await this.oracle.getConnection()
  try {
    const result = await conn.execute("SELECT 1 FROM DUAL",{ }, { outFormat:this.oracle.db.OBJECT })
    return result.rows
  } finally {
    await conn.close()
  }
}
```

```js
fastify.register(require("fastify-oracle"), {
pool: {
  user: "oracle",
  password: "oracle",
  connectString: "oracle.local:1521/mydb"
},
name: 'mydb'
}).register(require("fastify-oracle"), {
pool: {
  user: "oracle",
  password: "oracle",
  connectString: "oracle.local:1521/mydb"
},
name: 'otherdb'
})

async function (request, reply) {
  const myConn = await this.oracle.mydb.getConnection()
  const otherConn= await this.oracle.otherdb.getConnection()

  try {
    const myResult = await myConn.execute("SELECT 1 FROM DUAL",{ },{ outFormat: this.oracle.db.OBJECT })
    const otherResult = await otherConn.execute("SELECT 1 FROM DUAL",{ },{ outFormat: this.oracle.db.OBJECT})

    return {myResult, otherResult}
  } finally {
    await myConn .close()
    await otherConn.close()
  }
}
```

### CODE REVIEW

**Review by jsumners — COMMENTED** [review]
Documentation will need to be updated.

**Comment by cemremengu** [comment]
This PR ~no longer introduces a breaking change. It~ preserves and extends the previous API by exposing `getConnection()` method and `pool` object as individual properties. However, if you are using pool directly in your code, you will need to do `oracle.pool`

**Comment by cemremengu** [comment]
After spamming you gazillion of times (hopefully you unfollowed this PR), I think I am done 😄

**Comment by jsumners** [comment]
I just haven't had time to review it again. I'll get to it.

File: Readme.md [hunk 1]
```diff
@@ -53,6 +53,58 @@ takes precedence over the `pool` option.
 + `client`: an instance of an `oracledb` connection pool. This takes precedence
 over the `pool` and `poolAlias` options.
 
+A `name` option can be used in order to connect to multiple oracledb instances. 
+The first registered instance can be accessed via `fastify.oracle` or `fastify.oracle.<dbname>`. Note that once you register a named* instance, you will *not* be able to register an unnamed instance.
```
Comment (jsumners) [inline Readme.md]: There is a mis-matched `*` around `named*`.

**Pull request closed without merging**